emission/analysis/classification/inference/labels/pipeline.py

```diff
@@ -55,48 +57,68 @@ def run_prediction_pipeline(self, user_id, time_range):
         self.ts = esta.TimeSeries.get_time_series(user_id)
         self.toPredictTrips = esda.get_entries(
             esda.CLEANED_TRIP_KEY, user_id, time_query=time_range)
-        for cleaned_trip in self.toPredictTrips:
-            # Create an inferred trip
+        
+        cleaned_trip_list = self.toPredictTrips
+        inferred_trip_list = []
+        results_dict = {}
+        ensemble_list = []
+
+        # Create list of inferred trips
+        for cleaned_trip in cleaned_trip_list:
             cleaned_trip_dict = copy.copy(cleaned_trip)["data"]
             inferred_trip = ecwe.Entry.create_entry(user_id, "analysis/inferred_trip", cleaned_trip_dict)
-            
+            inferred_trip_list.append(inferred_trip)
+    
+        if inferred_trip_list:
```

Comment:
isn't this always going to be true, given that you define `inferred_trip_list` in line 62?
Are you actually testing against the "no trip" case?
I suspect this doesn't matter, since we just iterate through the trips downstream, but then we can just skip this check altogether given that it will always be true.

Reply:
I did test this out and Python evaluates an empty list to be False when used in an if condition.
So, it does not enter the if block unless trips are loaded into `inferred_trip_list`  in the for loop above the if block and the list becomes non-empty which then evaluates to True.